### PR TITLE
Add multihead matmul

### PR DESCRIPTION
**Pull request by Vimalakirti**

### CODE REVIEW

File: src/layer/matmul.rs [hunk 1]
```diff
@@ -42,9 +41,7 @@ impl Layer for MatMulLayer {
       }),
       N: 1,
     }));
-    let small_matmul = (a * b) < 1 << CONFIG.ptau.loaded_pow_len_log;
-    let use_cqlin = constants.len() > 1 && constants[1].is_some() && small_matmul;
-    let matmul_output = if use_cqlin {
+    let matmul_output = if constants.len() > 1 && constants[1].is_some() {
```
Comment (ddkang) [inline src/layer/matmul.rs]: What's going on here?

Comment (Vimalakirti) [reply src/layer/matmul.rs]: Good catch. I rebased on an incorrect commit. 
Now it's fixed.

**Comment by ddkang** [comment]
Can we make a test for this?

**Comment by Vimalakirti** [comment]
> Can we make a test for this?

We only have tests for basicblocks. If we want to test layers, we need to implement other logics to support it. But yes, we can test it later.

**Review by liliatangxy — APPROVED** [review]

**Review by ddkang — APPROVED** [review]

**Pull request merged**